### PR TITLE
Migrate Jetpack multisite callables from Sync package to Jetpack Plugin proper

### PR DESCRIPTION
**Pull request by mdbitz**
In #18797 we didn't migrate the multisite callables from the Sync package. This PR completes the migration so that Sync does not reference functionality not contained in its package or dependencies.

#### Changes proposed in this Pull Request:
* No functional changes, simply shifting addition of Jetpack multisite callables into Jetpack.

Does this pull request change what data or activity we track or use?
No
Testing instructions:
These changes are covered under the existing test test_sync_callable_whitelist

Manual Test
* yarn docker:wp shell --url=DOMAIN
* \Automattic\Jetpack\Sync\Defaults::$default_multisite_callable_whitelist; 
* Verify empty array returned
* \Automattic\Jetpack\Sync\Defaults::get_multisite_callable_whitelist();
* Verify network_name, network_allow_new_registrations, network_add_new_users, network_site_upload_space, network_upload_file_types and network_enable_administration_menus are returned. 

#### Proposed changelog entry for your changes:
* N/A - no functional changes simply shifting code :)

### CODE REVIEW

**Comment by anomiex** [comment]
Change itself looks ok. But it needs a master merge for the "Changelogger use" and "Changelogger validity" checks, and some changelogger change files so the former will pass.

File: projects/packages/sync/changelog/fix-extend-sync-multisite-callables [hunk 1]
```diff
@@ -0,0 +1,5 @@
+Significance: patch
+Type: changed
+Comment: jetpack Sync :: Migrate Jetpack multisite callables into the plugin vs defining them in the sync package. No functional change.
+
+
```
Comment (anomiex) [inline projects/packages/sync/changelog/fix-extend-sync-multisite-callables:5]: I'll leave it up to you, but so far for packages I've included every change (versus the "only user-facing changes" policy we follow for the Jetpack plugin).

**Review by anomiex — APPROVED** [review]
Haven't tested manually, but I see it's covered by an existing automated test.

One comment inline, but feel free to disagree.

File: projects/packages/sync/changelog/fix-extend-sync-multisite-callables (hunk 1 above)
Comment (mdbitz) [reply projects/packages/sync/changelog/fix-extend-sync-multisite-callables:5]: Yeah I'm on the fence with the packages. Sync isn't usable without Jetpack  at this time so the changelog entry at the package level isn't needed. 

I think when we get it fully separated then each change might be warranted so consumers are aware of all changes.

**Pull request merged**